src/firmware/mod.rs

```diff
@@ -114,6 +114,56 @@ pub enum Error {
     SecureDataInvalid,
 }
 
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let err_description = match self {
+            Error::IoError(_) => "I/O Error",
+            Error::InvalidPlatformState => "Invalid platform state",
+            Error::InvalidGuestState => "Invalid guest state",
+            Error::InvalidConfig => "Platform configuration invalid",
+            Error::InvalidLen => "Memory buffer to small",
+            Error::AlreadyOwned => "Platform is already owned",
+            Error::InvalidCertificate => "Invalid certificate",
+            Error::PolicyFailure => "Policy failure",
+            Error::Inactive => "Guest is inactive",
+            Error::InvalidAddress => "Provided address is invalid",
+            Error::BadSignature => "Provided signature is invalid",
+            Error::BadMeasurement => "Provided measurement is invalid",
+            Error::AsidOwned => "ASID is already owned",
+            Error::InvalidAsid => "ASID is invalid",
+            Error::WbinvdRequired => "WBINVD instruction required",
+            Error::DfFlushRequired => "`DF_FLUSH` invocation required",
+            Error::InvalidGuest => "Guest handle is invalid",
+            Error::InvalidCommand => "Issued command is invalid",
+            Error::Active => "Guest is active",
+            Error::HardwarePlatform => {
+                "Hardware condition occured, safe to re-allocate parameter buffers"
+            }
+            Error::HardwareUnsafe => {
+                "Hardware condition occured, unsafe to re-allocate parameter buffers"
+            }
```

Comment:
Just curious why the curly braces are used here but not for the others.

Reply:
`cargo fmt` placed those there, I assume because the original draft exceeded the 80 character line limit the Rust style suggests. If you think they're better off removed I can position everything on one line.

Reply:
No, I prefer whatever cargo fmt wants.